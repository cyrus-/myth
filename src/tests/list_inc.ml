let s = 
"type nat =
  | O
  | S of nat

type list =
  | Nil
  | Cons of nat * list

let rec map (l:list) (f : nat -> nat) : list =
     match l with
       | Nil -> Nil
       | Cons (n, ls) -> Cons (f n, map ls f)
;;

let list_inc : list -> list |>
  { [] => []
  | [1;2] => [2;3]
  | [0;0] => [1;1]
  | [3;4;5] => [4;5;6]
  } = ?"
