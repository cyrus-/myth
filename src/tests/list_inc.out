(* Generated Thu Nov 13 00:35:27 2014 *)
let list_inc : list -> list =
  fun (l1:list) -> map l1 (fun (n1:nat) -> S (n1))
;;
